agent/bench-scripts/pbench-fio

```diff
@@ -631,19 +666,9 @@ function fio_run_job() {
 	echo "running fio job: $fio_job_file ($(basename $benchmark_results_dir))"
 
 	if [ ! -z "${clients}" ]; then
-		debug_log "creating directories on the clients"
+		debug_log "killing any old fio processes, creating directories, and starting new a fio process on the clients"
 		for client in "${client_names[@]}"; do
-			ssh ${ssh_opts} ${client} mkdir -p ${benchmark_results_dir} &
-		done
-		wait
-		debug_log "killing any old fio process on the clients"
-		for client in "${client_names[@]}"; do
-			ssh ${ssh_opts} ${client} "killall fio >/dev/null 2>&1" &
-		done
-		wait
-		debug_log "starting new fio process on the clients"
-		for client in "${client_names[@]}"; do
-			ssh ${ssh_opts} ${client} "pushd ${benchmark_results_dir} >/dev/null; screen -dmS fio-server bash -c ''$benchmark_bin' --server=,${client_ports[${client}]} 2>&1 >client-result.txt'"
+			ssh ${ssh_opts} ${client} "killall ${benchmark} >/dev/null 2>&1; mkdir -p ${benchmark_results_dir}; pushd ${benchmark_results_dir} >/dev/null; screen -dmS fio-server bash -c ''$benchmark_bin' --server=,${client_ports[${client}]} 2>&1 >client-result.txt'"
```

Comment:
And the more I look at the double-single quote thing the more I don't understand what it is doing.

Reply:
> And the more I look at the double-single quote thing the more I don't understand what it is doing.

Especially the empty `''` block. The second single-quote would seem to protect the variables from expansion here, deferring to the sub-bash under the screen (weird enough already); but I'm not sure I understand why since if they have any value at all in that context (i.e., they're exported ... are they?), there's no mechanism for the expansion to differ in that context. Bee-zarre.